### ISSUE TITLE
While using HiddenField, ignoreSetter fails if setter returns a parameterized class

### ISSUE THREAD

**Issue by alugan** [checkstyle/checkstyle#8770]
A configuration such as :
```
<module name="HiddenField" >
    <property name="ignoreSetter" value="true" />
    <property name="ignoreConstructorParameter" value="true" />
</module>
```
still raises errors with the following code
```
public class AbstractMyClass {

private String myAttribute;

public <T extends AbstractMyClass> T setMyAttribute(String myAttribute) {
        this.myAttribute = myAttribute;
        return (T) this;
    }
}
```

---------------
The setMyAttribute() method should be recognized as a setter
--------------

**Comment by pbludov** [comment]
@alugan please give us more details on this issue. The bug report template can be found here:
https://checkstyle.org/report_issue.html#How_to_report_a_bug.3F

**Comment by alugan** [comment]
> @alugan please give us more details on this issue. The bug report template can be found here:
> https://checkstyle.org/report_issue.html#How_to_report_a_bug.3F

@pbludov I complete the code so you could copy/paste the entire files into any java project.

**Comment by pbludov** [comment]
> @pbludov I complete the code so you could copy/paste the entire files into any java project.

No, you don't. You forgot to provide a link to the documentation and output of the program. 

The important point here is that Checkstyle is not a type-aware tool. It does not contain code to infer that `T` is `AbstractMyClass`. In this particular case, it is possible to implement such algorithm, but in general case it may not work.

**Comment by romani** [comment]
> I complete the code so you could copy/paste the entire files into any java project.

Please strictly follow template, and show exact output of CLI.
